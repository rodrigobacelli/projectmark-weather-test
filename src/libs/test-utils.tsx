import * as React from 'react';
import { render, RenderOptions } from '@testing-library/react';

const Providers = ({ children }: { children: React.ReactNode }) => {
  return children;
};

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { customRender as render };
