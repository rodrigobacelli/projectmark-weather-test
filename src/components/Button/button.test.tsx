import { describe, it, expect, vi } from 'vitest';
import { render } from '../../tests/test-utils';
import { Button } from './';

describe('components/Button', () => {
  it('should render the Button Label', () => {
    const result = render(<Button>Label</Button>);

    expect(result.getByText('Label')).toBeInTheDocument();
  });

  it('should render the Button StartIcon', () => {
    const result = render(
      <Button startIcon={() => <div data-testid="startIcon" />}>Label</Button>
    );

    expect(result.getByTestId('startIcon')).toBeInTheDocument();
  });

  it('should render the Button EndIcon', () => {
    const result = render(
      <Button endIcon={() => <div data-testid="endIcon" />}>Label</Button>
    );

    expect(result.getByTestId('endIcon')).toBeInTheDocument();
  });

  it('should accept classNames', () => {
    const result = render(<Button className="button-className">Label</Button>);

    expect(
      result.container.querySelector('.button-className')
    ).toBeInTheDocument();
  });

  it('should handle Button click', async () => {
    const onClick = vi.fn();
    const result = render(<Button onClick={onClick}>Label</Button>);

    await result.getByRole('button').click();

    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('should render the Button Component', () => {
    const result = render(
      <Button
        endIcon={() => <div data-testid="endIcon" />}
        startIcon={() => <div data-testid="startIcon" />}
        className="button-className"
      >
        Label
      </Button>
    );

    expect(result).toMatchSnapshot();
  });
});
