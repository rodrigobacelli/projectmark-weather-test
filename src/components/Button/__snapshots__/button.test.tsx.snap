// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should render the Button Component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="text-sm flex items-center gap-1 px-2 py-1 rounded-sm border enabled:cursor-pointer text-gray-900 dark:text-gray-300 border-gray-900 dark:border-gray-300 enabled:hover:bg-gray-900/20 enabled:dark:hover:bg-gray-100/20 enabled:focus:bg-gray-800/10 enabled:dark:focus:bg-gray-200/10 disabled:text-gray-500 disabled:border-gray-500 button-className"
      >
        <div
          data-testid="startIcon"
        />
        Label
        <div
          data-testid="endIcon"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="text-sm flex items-center gap-1 px-2 py-1 rounded-sm border enabled:cursor-pointer text-gray-900 dark:text-gray-300 border-gray-900 dark:border-gray-300 enabled:hover:bg-gray-900/20 enabled:dark:hover:bg-gray-100/20 enabled:focus:bg-gray-800/10 enabled:dark:focus:bg-gray-200/10 disabled:text-gray-500 disabled:border-gray-500 button-className"
    >
      <div
        data-testid="startIcon"
      />
      Label
      <div
        data-testid="endIcon"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/Button > should render the Button Component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="text-sm flex items-center gap-1 px-2 py-1 rounded-sm border enabled:cursor-pointer text-gray-900 dark:text-gray-300 border-gray-900 dark:border-gray-300 enabled:hover:bg-gray-900/20 enabled:dark:hover:bg-gray-100/20 enabled:focus:bg-gray-800/10 enabled:dark:focus:bg-gray-200/10 disabled:text-gray-500 disabled:border-gray-500 button-className"
      >
        <div
          data-testid="startIcon"
        />
        Label
        <div
          data-testid="endIcon"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="text-sm flex items-center gap-1 px-2 py-1 rounded-sm border enabled:cursor-pointer text-gray-900 dark:text-gray-300 border-gray-900 dark:border-gray-300 enabled:hover:bg-gray-900/20 enabled:dark:hover:bg-gray-100/20 enabled:focus:bg-gray-800/10 enabled:dark:focus:bg-gray-200/10 disabled:text-gray-500 disabled:border-gray-500 button-className"
    >
      <div
        data-testid="startIcon"
      />
      Label
      <div
        data-testid="endIcon"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
